vendor/riscv/riscv-isa-sim/riscv/Proc.cc

```diff
@@ -207,42 +203,41 @@ void Processor::take_trap(trap_t &t, reg_t epc) {
              DEFAULT_PRIV, "Privilege Level");
-  params.set(base, "boot_addr", any(0x80000000UL), "uint64_t", "0x80000000UL",
+  params.set_uint64_t(base, "boot_addr", 0x80000000UL, "0x80000000UL",
              "First PC of the core");
-  params.set(base, "mmu_mode", any(std::string("sv39")), "string", "sv39",
+  params.set_string(base, "mmu_mode", "sv39", "sv39",
              "Memory virtualization mode");
 
-  params.set(base, "pmpregions", std::any(0x0UL), "uint64_t", "0x0",
+  params.set_uint64_t(base, "pmpregions", 0x0UL, "0x0",
              "Number of PMP regions");
-  params.set(base, "pmpaddr0", any(0x0UL), "uint64_t", "0x0",
+  params.set_uint64_t(base, "pmpaddr0", 0x0UL, "0x0",
              "Default PMPADDR0 value");
-  params.set(base, "pmpcfg0", any(0x0UL), "uint64_t", "0x0",
+  params.set_uint64_t(base, "pmpcfg0", 0x0UL, "0x0",
              "Default PMPCFG0 value");
-  params.set(base, "marchid", any(0x3UL), "uint64_t", "0x3", "MARCHID value");
-  params.set(base, "mhartid", any(0x0UL), "uint64_t", "0x0", "MHARTID value");
-  params.set(base, "mvendorid", any(0x00000602UL), "uint64_t", "0x00000602UL",
+  params.set_uint64_t(base, "marchid", 0x3UL, "0x3", "MARCHID value");
+  params.set_uint64_t(base, "mhartid", 0x0UL, "0x0", "MHARTID value");
+  params.set_uint64_t(base, "mvendorid", 0x00000602UL, "0x00000602UL",
              "MVENDORID value");
-  params.set(base, "extensions", any(std::string("")), "string",
-             "\"extension1,extension2\"", "Possible extensions: cv32a60x");
+  params.set_string(base, "extensions", "", "", "Possible extensions: cv32a60x, cvxif");
 
-  params.set(base, "status_fs_field_we_enable", any(false), "bool", "false",
+  params.set_bool(base, "status_fs_field_we_enable", false, "false",
              "XSTATUS CSR FS Write Enable param enable");
-  params.set(base, "status_fs_field_we", any(false), "bool", "false",
+  params.set_bool(base, "status_fs_field_we", false, "false",
              "XSTATUS CSR FS Write Enable");
-  params.set(base, "status_vs_field_we_enable", any(false), "bool", "false",
+  params.set_bool(base, "status_vs_field_we_enable", false, "false",
              "XSTATUS CSR VS Write Enable param enable");
-  params.set(base, "status_vs_field_we", any(false), "bool", "false",
+  params.set_bool(base, "status_vs_field_we", false, "false",
              "XSTATUS CSR VS Write Enable");
-  params.set(base, "misa_we_enable", any(true), "bool", "true",
+  params.set_bool(base, "misa_we_enable", true, "true",
              "MISA CSR Write Enable param enable");
-  params.set(base, "misa_we", any(false), "bool", "false",
+  params.set_bool(base, "misa_we", false, "false",
              "MISA CSR Write Enable value");
 
-  params.set(base, "misaligned", std::any(false), "bool", "false",
+  params.set_bool(base, "misaligned", false, "false",
              "Support for misaligned memory operations");
 
   params.set(base, "csr_counters_injection", std::any(false), "bool", "false",
```

Comment:
This should probably be : 
`params.set_bool(base, "csr_counters_injection", false, "false",
             "Allow to set CSRs getting values from a DPI");`
(Will not build otherwise)